### ISSUE TITLE
Petition to add commutative algebra

### ISSUE THREAD

**Issue by Admiraldesvl** [ossu/math#13]
Commutative algebra is the study of commutative ring and is a ticket to a lot of modern mathematics. For example, homological algebra, algebraic number theory, algebraic geometry, arithmetic geometry, Diophantine geometry. It is impossible for a algebra-focused student to continue without the study of commutative algebra. This is the reason I open this issue.

**Which year and prerequisite?**

In the book _Introduction to Commutative Algebra_ by Michael Atiyah and I. G. MacDonald, the introduction said

> This book grew out of a course of lectures given to third year undergraduates at Oxford University and it has the modest aim of providing a rapid introduction to the subject. It is designed to be read by students who have had a first elementary course in general algebra.

**Resources**

One of the most classics out there is _Introduction to Commutative Algebra_ by Michael Atiyah and I. G. MacDonald. I have found PDF file [here](http://math.univ-lyon1.fr/~mathieu/CoursM2-2020/AMD-ComAlg.pdf). It was uploaded publicly by a professor from Université Claude Bernard Lyon 1, so it's not likely to be some violation of copyright (If it is piracy let's find equivalences somewhere else). There is a discussion on the prerequisite of this book on [stackexchange](https://math.stackexchange.com/questions/60831/prerequisites-for-atiyah-macdonald/60840)

There is a list that strictly follow the book mentioned above (final two chapters are missing though): https://www.youtube.com/watch?v=VKxT2lkmMVE&list=PLq-Gm0yRYwTjBziGqSW9kFF9o2l5ECDvY

Another list but it follows a different book ([Commutative Algebra - with a View Toward Algebraic Geometry](https://link.springer.com/book/10.1007/978-1-4612-5350-1)): https://www.youtube.com/watch?v=QOTf8KfrZFU&list=PL8yHsr3EFj53rSexSz7vsYt-3rpHPR3HB

**Comment by waciumawanjohi** [comment]
@Admiraldesvl 
These seem like a number of resources that can support the study of commutative algebra. What great research on your part!

You detail what further fields can be studied after commutative algebra. Can you flesh this out a little more? Are you advocating that all OSSUnians studying math should be directed to commutative algebra? Or that this should be an optional course that has later follow on courses?

As context, [OSSU:CS](https:cs.ossu.dev) is our original curriculum. In it, there are a core set of courses that are recommended to everyone and a number of advanced tracks where students are expected to choose one or more. My hypothesis is that the math curriculum will be similar (but I would not dictate that structure).

**Comment by Admiraldesvl** [comment]
> Are you advocating that all OSSUnians studying math should be directed to commutative algebra? Or that this should be an optional course that has later follow on courses?

It's optional in general. If a student wants to focus on algebra-related area seriously, then commutative algebra is nearly a must everywhere.

**Comment by waciumawanjohi** [comment]
Under [Advanced Topics](https://github.com/ossu/math#advanced-topics), there is [Abstract Algebra](https://github.com/ossu/math#abstract-algebra) (which is in need of fleshing out). Would commutative algebra belong here, or should it be in a different advanced topic?

**Comment by aayushsinha0706** [comment]
Yes please add commutative algebra under abstract algebra. Also Commutative Algebra is a recommended pre-requisite of algebraic geometry

**Comment by Admiraldesvl** [comment]
I think it is OK to put commutative algebra under abstract algebra. One may treat it as Abstract Algebra II or III. I found another lecture note: [Lecture note on algebra 3](https://wwli.asia/downloads/YAlg3.pdf).

You can keep the discussion as you want and I will try to give what I can.

**Comment by aayushsinha0706** [comment]
I guess two playlist are enough for Commutative Algebra.

**Comment by Admiraldesvl** [comment]
There is another one: https://www.youtube.com/watch?v=ZgwJdT0DaQ4&list=PLn6dA-hP_G8SR-v8EV5m9vcpo-V9No-2V

**Comment by aayushsinha0706** [comment]
Well in my issue that I have taken the course that rigorously follows the text book.

**Comment by Admiraldesvl** [comment]
I have only found one playlist that use Atiyah-MacDonald strictly, though final two chapters are missing. I think it is quite rare to find some other playlists that still meet this criteria. The recent one does not specify the text it follows, but I think it shouldn't be a big problem (as long as it is self-contained). By the way when I was taking this course in university, our professor did not strictly follow one single book as well.

**Comment by aayushsinha0706** [comment]
Yeah. I have recommended a set of courses for Advanced Topics after investigating a lot in issue#14. I hope they see into it.
